init/eessi_environment_variables

```diff
@@ -25,9 +25,16 @@ if [ -d $EESSI_PREFIX ]; then
   if [ -d $EESSI_EPREFIX ]; then
 
     # determine subdirectory in software layer
-    # note: eessi_software_subdir_for_host.py will pick up value from $EESSI_SOFTWARE_SUBDIR_OVERRIDE if it's defined!
-    export EESSI_EPREFIX_PYTHON=$EESSI_EPREFIX/usr/bin/python3
-    export EESSI_SOFTWARE_SUBDIR=$($EESSI_EPREFIX_PYTHON ${EESSI_INIT_DIR_PATH}/eessi_software_subdir_for_host.py $EESSI_PREFIX)
+    if [ -z $EESSI_USE_ARCHDETECT ]; then
```

Comment:
@hmeiland The condition is wrong here: currently, if `$EESSI_USE_ARCHDETECT` is set (then `-z` is false), you're using `archspec` rather the `archdetect` as intended...

Let's use a more explicit check, and only enable archdetect when `$EESSI_USE_ARCHDETECT` is set to `1`:

```suggestion
    if [ "$EESSI_USE_ARCHDETECT" == "1" ]; then
```

We should also make the archdetect mode honor `$EESSI_SOFTWARE_SUBDIR_OVERRIDE` btw, but that can be done in a follow-up PR